mlos_bench/mlos_bench/services/types/vm_provisioner_type.py

```diff
@@ -72,32 +72,47 @@ def vm_start(self, params: dict) -> Tuple["Status", dict]:
             Status is one of {PENDING, SUCCEEDED, FAILED}
         """
 
-    def vm_stop(self) -> Tuple["Status", dict]:
+    def vm_stop(self, params: dict) -> Tuple["Status", dict]:
         """
         Stops the VM by initiating a graceful shutdown.
 
+        Parameters
+        ----------
+        params : dict
+            Flat dictionary of (key, value) pairs of tunable parameters.
+
         Returns
         -------
         result : (Status, dict={})
             A pair of Status and result. The result is always {}.
             Status is one of {PENDING, SUCCEEDED, FAILED}
         """
 
-    def vm_restart(self) -> Tuple["Status", dict]:
+    def vm_restart(self, params: dict) -> Tuple["Status", dict]:
```

Comment:
`params` is a flat `dict` of *all* the tunables (for that env), global configs, const args, etc.
That could be a lot of things.
Some of which may involve some things that aren't excepted by the Service that implements this since it may have different semantics for different systems.  Just wondering if we're going to have to filter this in the future ...

Reply:
yes, we will definitely need to filter those, I just have to think of an elegant way to do it. Will do in the subsequent PRs